public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,740 @@
+import { FeatureAccordion } from '../../../../public/pages/ConfigureModel/components/FeatureAccordion';
+import {
+  AD_DOCS_LINK,
+  AD_HIGH_CARDINALITY_LINK,
+  MAX_FEATURE_NUM,
+} from '../../../../public/utils/constants';
+import { getNotifications } from '../../../../public/services';
+import { prettifyErrorMessage } from '../../../../server/utils/helpers';
+
+function AddAnomalyDetector({ embeddable, closeFlyout, mode, setMode }) {
+  const dispatch = useDispatch();
+  const [queryText, setQueryText] = useState('');
+  useEffect(() => {
+    const getInitialIndices = async () => {
+      await dispatch(getIndices(queryText));
+    };
+    getInitialIndices();
+    dispatch(getMappings(embeddable.vis.data.aggs.indexPattern.title));
+  }, []);
+
+  const [isShowVis, setIsShowVis] = useState(false);
+  const [accordionsOpen, setAccordionsOpen] = useState({ modelFeatures: true });
+  const [detectorNameFromVis, setDetectorNameFromVis] = useState(
+    formikToDetectorName(embeddable.vis.title)
+  );
+  const [intervalValue, setIntervalalue] = useState(10);
+  const [delayValue, setDelayValue] = useState(1);
+  const [enabled, setEnabled] = useState<boolean>(false);
+
+  const title = embeddable.getTitle();
+  const onAccordionToggle = (key) => {
+    const newAccordionsOpen = { ...accordionsOpen };
+    newAccordionsOpen[key] = !accordionsOpen[key];
+    setAccordionsOpen(newAccordionsOpen);
+  };
+  const onDetectorNameChange = (e, field) => {
+    field.onChange(e);
+    setDetectorNameFromVis(e.target.value);
+  };
+  const onIntervalChange = (e, field) => {
+    field.onChange(e);
+    setIntervalalue(e.target.value);
+  };
+  const onDelayChange = (e, field) => {
+    field.onChange(e);
+    setDelayValue(e.target.value);
+  };
+  const aggList = embeddable.vis.data.aggs.aggs.filter(
+    (feature) => feature.schema == 'metric'
+  );
+  const featureList = aggList.filter(
+    (feature, index) =>
+      index <
+      (aggList.length < MAX_FEATURE_NUM ? aggList.length : MAX_FEATURE_NUM)
+  );
+
+  const getFeatureNameFromVisParams = (id) => {
+    let name = find(embeddable.vis.params.seriesParams, function (param) {
+      if (param.data.id === id) {
+        return true;
+      }
+    });
+
+    return name.data.label.replace(/[^a-zA-Z0-9-_]/g, '_');
+  };
+
+  const notifications = getNotifications();
+
+  const handleValidationAndSubmit = (formikProps) => {
+    if (!isEmpty(formikProps.errors)) {
+      focusOnFirstWrongFeature(formikProps.errors, formikProps.setFieldTouched);
+      notifications.toasts.addDanger('One or more input fields is invalid');
+    } else {
+      handleSubmit(formikProps);
+    }
+  };
+
+  const handleSubmit = (formikProps) => {
+    formikProps.setSubmitting(true);
+    try {
+      const detectorToCreate = formikToDetector(formikProps.values);
+      dispatch(createDetector(detectorToCreate))
+        .then(async (response) => {
+          notifications.toasts.addSuccess(
```

Comment:
We have different notification messages in the mock ups, we should probably align with what is there and/or bring up with UX if we want to make any changes

Reply:
Will do as part of the alerting integration